scrape/scrape.go

```diff
@@ -1380,14 +1394,18 @@ func (sl *scrapeLoop) scrapeAndReport(last, appendTime time.Time, errc chan<- er
 
 	// A failed scrape is the same as an empty scrape,
 	// we still call sl.append to trigger stale markers.
+	targetScrapeAppendAttemptsTotal.Inc()
 	total, added, seriesAdded, appErr = sl.append(app, b, contentType, appendTime)
 	if appErr != nil {
+		targetScrapeAppendErrorsTotal.Inc()
 		app.Rollback()
 		app = sl.appender(sl.appenderCtx)
 		level.Debug(sl.l).Log("msg", "Append failed", "err", appErr)
 		// The append failed, probably due to a parse error or sample limit.
 		// Call sl.append again with an empty scrape to trigger stale markers.
+		targetScrapeAppendAttemptsTotal.Inc()
```

Comment:
Does including retries really add much value here? You could have just incremented the errors if both attempts failed. With retries you're skewing your total and making your failure rate look worse.

For example, let's say 1 goes through the first time, 1 goes through the second time and 1 goes through neither time.
Without retries, you get: total=3, errors=1, failure ratio=0.33%
With retries, you get: total=5, errors=3, failure ratio=0.6%

Reply:
Good point, while implementing I was not sure what approach to take here. Sounds good to increment once for total attempts and only increment errors if even the retry fails

Reply:
Thanks! Looks good to me!